casbin/enforcer.cpp

```diff
@@ -162,9 +162,9 @@ bool Enforcer :: enforce(string matcher, Scope scope) {
 /**
  * Enforcer is the default constructor.
  */
-Enforcer* Enforcer :: NewEnforcer() {
-    Enforcer* e = new Enforcer;
-    return e;
+unique_ptr<Enforcer> Enforcer ::NewEnforcer() {
```

Comment:
And also @ZipoChan can you replace this class methods to C++ constructors instead.

Reply:
I think, this changes can be made in another PR.

Reply:
If using a constructor, then don't return a pointer, return the object is ok and use a pointer if want to share in other places.